src/main/scala/is/hail/methods/PCA.scala

```diff
@@ -78,8 +75,7 @@ object PCA {
           s"but user requested ${ k } principal components.")
 
     val optionLoadings = someIf(computeLoadings, {
-      val rowType = TStruct("v" -> vsm.rowKeyStruct) ++ (if (asArray) TStruct("loadings" -> pcSchema(k, asArray))
-      else pcSchema(k, asArray).asInstanceOf[TStruct])
+      val rowType = TStruct("v" -> vsm.rowKeyStruct) ++ (if (asArray) TStruct("loadings" -> TArray(TFloat64())) else pcSchema(k))
```

Comment:
shouldn't we explode the row key struct? It could cause problems if it has fields named "PCx", though.

Reply:
yeah, this is already done in the sample scores table above.